### ISSUE TITLE
Optics-based Bubble Tracking and Velocity Measurement

### ISSUE THREAD

**Issue by djcranshaw** [picoexperiment/AutoBub3hs#10]
By analysing the camera images several frames after the bubbles in an event are first detected, it should be possible to obtain position measurements of the bubble as a function of frame, and therefore a velocity measurement of the initial bubble. Doing this could allow us to study convection within the C3F8. Doing this will may require modifications to Autobub or XYZLookup or both to allow the detection and position reconstruction of bubbles after they have grown larger than a few pixels.

**Comment by djcranshaw** [comment]
Some updates:
2024/08/14: https://www.snolab.ca/pico-docdb/cgi/ShowDocument?docid=5853

**Comment by jeremysavoie** [comment]
As far as I know, Autobub is able to find the velocity of single bubble events. Is this issue more related towards improving its ability to do that or more about studying the convection within the detector? I was wondering what was the goal of this issue especially considering the update. We could either move it to the analysis repository under Convection study, or close this issue.

**Comment by djcranshaw** [comment]
You're correct, I wasn't aware Autobub could already do this when I wrote this up. This would be more appropriate in the analysis repo, with a slightly different name and description as you suggest, since the current task is really using the tracking information to study convection (which is why I assigned Quinn to it recently). Since there is already an update linked to it and I'd like to preserve that timeline, I think I slighly prefer transfer-and-rename to close-and-open-new, but the preference isn't strong.
